src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -7,47 +7,56 @@
 import java.io.FileWriter;
 
 public class WorkWithFile {
+    private StringBuilder stringBuilder = new StringBuilder();
 
     public void getStatistic(String fromFileName, String toFileName) {
-        StringBuilder stringBuilder = new StringBuilder();
+        String[] dataFromMarket = readData(fromFileName);
+        String report = processData(dataFromMarket);
+        createReport(toFileName, report);
+    }
+
+    private String[] readData(String fromFileName) {
         try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
             String value = bufferedReader.readLine();
             while (value != null) {
                 stringBuilder.append(value).append(System.lineSeparator());
                 value = bufferedReader.readLine();
             }
         } catch (Exception e) {
-            throw new RuntimeException("I can't read read file" + toFileName, e);
-        }
-        String[] dataFromMarket = stringBuilder.toString().split(System.lineSeparator());
-        String report = createReport(dataFromMarket);
-        try {
-            File toFilFile = new File(toFileName);
-            toFilFile.createNewFile();
-        } catch (Exception e) {
-            throw new RuntimeException("I can't create file" + toFileName, e);
-        }
-        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
-            bufferedWriter.write(report);
-        } catch (Exception e) {
-            throw new RuntimeException("I can't write information to the file" + toFileName, e);
+            throw new RuntimeException("I can't read read file" + fromFileName, e);
         }
+        return stringBuilder.toString().split(System.lineSeparator());
     }
 
-    public String createReport(String[] dataFromMarket) {
+    private String processData(String[] dataFromMarket) {
+        stringBuilder.setLength(0);
         int supply = 0;
         int buy = 0;
         for (int i = 0; i < dataFromMarket.length; i++) {
             if (dataFromMarket[i].contains("supply")) {
                 supply += Integer.parseInt(dataFromMarket[i].substring(dataFromMarket[i]
-                        .indexOf(",") + 1, dataFromMarket[i].length()));
+                        .indexOf(",") + 1));
             } else if (dataFromMarket[i].contains("buy")) {
                 buy += Integer.parseInt(dataFromMarket[i].substring(dataFromMarket[i]
-                        .indexOf(",") + 1, dataFromMarket[i].length()));
+                        .indexOf(",") + 1));
             }
         }
-        return "supply," + supply + System.lineSeparator()
-                + "buy," + buy + System.lineSeparator()
-                + "result," + (supply - buy);
+        return stringBuilder.append("supply,").append(supply).append(System.lineSeparator())
+                .append("buy,").append(buy).append(System.lineSeparator())
+                .append("result,").append(supply - buy).toString();
```

Comment:
counting stock and creating report are different actions. lets divide this method in two: `countData` and `createReport`